src/AppInstallerCommonCore/Public/winget/GroupPolicy.h

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+
+#include "AppInstallerLanguageUtilities.h"
+#include "winget/Registry.h"
+#include <string_view>
+
+namespace AppInstaller::Settings
+{
+    using namespace std::string_view_literals;
+
+    // A policy that sets a value for some setting.
+    // The value of the policy is a value in the registry key, or is
+    // made up of sub-keys for settings that are lists.
+    enum class ValuePolicy
+    {
+        SourceAutoUpdateIntervalInMinutes,
+        IncludeSources, // TODO
+        Max,
+    };
+
+    // A policy that acts as a toggle to enable or disable a feature.
+    // They are backed by a DWORD value with values 0 and 1.
+    enum class TogglePolicy
+    {
+        None = 0,
+        WinGet,
+        SettingsCommand,
+        ExperimentalFeatures,
+        LocalManifestFiles,
+        DefaultSource,
+        MSStoreSource,
+        AdditionalSources,
+        AllowedSources,
+        Max,
+    };
+
+    // Possible configuration states for a policy.
+    enum class PolicyState
+    {
+        NotConfigured,
+        Disabled,
+        Enabled,
+    };
+
+    namespace details
+    {
+
+        template <ValuePolicy P>
+        struct ValuePolicyMapping
+        {
+            // value_t - type of the policy
+            // ReadAndValidate() - Function that reads the value and does semantic validation.
+
+            // For simple values:
+            //  ValueName - Name of the registry value
+            //  ValueType - Type of the registry value
+            //  reg_value_t - Type returned by the registry when reading the value
+        };
+
+#define POLICYMAPPING_SPECIALIZATION(_policy_, _type_) \

```

Comment:
Do you need to write this as one word instead of:
```suggestion
#define POLICY_MAPPING_SPECIALIZATION(_policy_, _type_) \
```

?

Reply:
I used one word to mirror the existing SETTINGMAPPING_SPECIALIZATION, but I do like it more as two words. I will change it to two.